lib/codegen/src/dbg.rs

```diff
@@ -1,132 +1,21 @@
-
-/// Is debug tracing enabled?
-///
-/// Debug tracing can be enabled by setting the `CRANELIFT_DBG` environment variable to something
-/// other than `0`.
-///
-/// This inline function turns into a constant `false` when debug assertions are disabled.
-#[cfg(feature = "std")]
-#[inline]
-pub fn enabled() -> bool {
-    if cfg!(debug_assertions) {
-        match STATE.load(atomic::Ordering::Relaxed) {
-            0 => initialize(),
-            s => s > 0,
-        }
-    } else {
-        false
-    }
-}
-
-/// Does nothing
-#[cfg(not(feature = "std"))]
-#[inline]
-pub fn enabled() -> bool {
-    false
-}
-
-/// Initialize `STATE` from the environment variable.
-#[cfg(feature = "std")]
-fn initialize() -> bool {
-    let enable = match env::var_os("CRANELIFT_DBG") {
-        Some(s) => s != OsStr::new("0"),
-        None => false,
-    };
 
-    if enable {
-        STATE.store(1, atomic::Ordering::Relaxed);
-    } else {
-        STATE.store(-1, atomic::Ordering::Relaxed);
-    }
-
-    enable
-}
-
-#[cfg(feature = "std")]
-thread_local! {
-    static WRITER : RefCell<io::BufWriter<File>> = RefCell::new(open_file());
-}
-
-/// Write a line with the given format arguments.
-///
-/// This is for use by the `dbg!` macro.
-#[cfg(feature = "std")]
-pub fn writeln_with_format_args(args: fmt::Arguments) -> io::Result<()> {
-    WRITER.with(|rc| {
-        let mut w = rc.borrow_mut();
-        writeln!(*w, "{}", args)?;
-        w.flush()
-    })
-}
-
-/// Open the tracing file for the current thread.
-#[cfg(feature = "std")]
-fn open_file() -> io::BufWriter<File> {
-    let curthread = thread::current();
-    let tmpstr;
-    let mut path = "cranelift.dbg.".to_owned();
-    path.extend(
-        match curthread.name() {
-            Some(name) => name.chars(),
-            // The thread is unnamed, so use the thread ID instead.
-            None => {
-                tmpstr = format!("{:?}", curthread.id());
-                tmpstr.chars()
-            }
-        }.filter(|ch| ch.is_alphanumeric() || *ch == '-' || *ch == '_'),
-    );
-    let file = File::create(path).expect("Can't open tracing file");
-    io::BufWriter::new(file)
-}
-
-/// Write a line to the debug trace file if tracing is enabled.
-///
-/// Arguments are the same as for `printf!`.
+/// Initialize logging state from the environment variable.
 #[cfg(feature = "std")]
-#[macro_export]
-macro_rules! dbg {
-    ($($arg:tt)+) => {
-        if $crate::dbg::enabled() {
-            // Drop the error result so we don't get compiler errors for ignoring it.
-            // What are you going to do, log the error?
-            $crate::dbg::writeln_with_format_args(format_args!($($arg)+)).ok();
-        }
-    }
-}
-
-/// `dbg!` isn't supported in `no_std` mode, so expand it into nothing.
-#[cfg(not(feature = "std"))]
-#[macro_export]
-macro_rules! dbg {
-    ($($arg:tt)+) => {};
+pub fn setup_logging() {
+    cranelift_logger::initialize("cranelift.dbg.");
 }
```

Comment:
If I understand what's going on here, this is the only place where cranelift_codegen depends on cranelift_logger explicitly. Is there a way we can move this, to avoid that dependency?

Reply:
That's correct. This function is used in 3 external locations (2 when creating new threads, in particular, which need to set up their TLS state), so that was one way to avoid leaking the filename prefix `cranelift.dbg`. We could just not care too much about uniformity in naming and have the callers directly call `cranelift_logger::initialize`, if you'd prefer. Alternatively, if there's an initialization function anywhere in any cranelift crate, we could use that instead.

Reply:
What if we define the string `cranelift.dbg` in cranelift-codegen and export it, so callers can call `cranelift_logger::initialize` themselves without hard-coding the string?